### DISCUSSION TITLE
Call to a member function keyBy() on null on implementing Cluster Functionality

### DISCUSSION THREAD

**Discussion by octaviogamero27** [filamentphp/filament#10856] [Help] [answered]
I have this issue when implemented Cluster for new version V3.2.2, when I set one resource to the Cluster, that Its has Five Resources on it.
Please help me to resolve this issue, Thanks a lot 
`
    protected static ?string $model = Solicitudes::class;

    protected static ?string $navigationIcon = 'heroicon-o-document-check';

    protected static ?string $label = "Solicitudes";
    protected static ?string $cluster = Clientes::class; //Cluster Class`

![image](https://github.com/filamentphp/filament/assets/43446223/b14e6775-f01d-4923-af23-810d4a34ccf6)

![image](https://github.com/filamentphp/filament/assets/43446223/1c9e5097-4641-4358-9b29-4497555a90b6)

**Comment by ayatamine** [comment]
hi, did you find any solution?

**Comment by bmpk99** [comment, marked as answer]
Yes, i've found the solution :)

So basically, this issue is related to parent/child navigation resources in my case i had a resource named "ProjectResource" where i was trying to change the navigation group using getNavigationGroup() method in the resource which would not let me change and output the same error and it was because i had another resource named "CategoryResource" whhich was the child navigation resource of the "ProjectResource" and in order to overcome this issue i had to keep the same getNavigationGroup() returns in both resources.

in short, make sure all child resources have the same getNavigationGroup() returns.

Hope it helps!

**Reply by octaviogamero27** [reply]
THANKS! For answer me!!!
I'm gonna review my code to fix all that stuff.

Greetings!!🙏🏻
